File: apps/web/lib/types/packet.ts
import { Message } from "./message";

/**
 * Packets and signals sent & received to/from server-side to keep track of room status
 */

export enum ClientSendSignal {}

export enum ServerSendSignal {
  TRANSCRIPT_UPDATE = "transcript_update",
  CLONE_SPEECH_END = "clone_speech_end",
  CONVERSATION_END = "conversation_end",
}

export interface Packet {
  signal: ClientSendSignal | ServerSendSignal;
  message?: Message;
}

File: apps/fastapi/app/lib/types/packet.py
from enum import Enum
from typing import Optional, Union

from app.lib.types.message import Message
from pydantic import BaseModel

"""Packets and signals sent & received to/from client-side to keep track of room status"""


class ClientSendSignal(Enum):
    GUEST_JOINED_SUCCESS = "GUEST_JOINED_SUCCESS"
    GUEST_SPEECH_END = "GUEST_SPEECH_END"


class ServerSendSignal(Enum):
    GUEST_JOINED_SUCCESS_ACK = "GUEST_JOINED_SUCCESS_ACK"
    TRANSCRIPT_UPDATE = "TRANSCRIPT_UPDATE"
    CLONE_SPEECH_END = "CLONE_SPEECH_END"
    CONVERSATION_END = "CONVERSATION_END"


class Packet(BaseModel):
    signal: Union[ClientSendSignal, ServerSendSignal]
    message: Optional[Message] = None
